pkg/grpc/server.go

```diff
@@ -335,12 +347,18 @@ func (a *apiImpl) muxer(localConn *grpc.ClientConn) http.Handler {
 	return mux
 }
 
+func (a *apiImpl) stop(finishedSig *concurrency.Signal) {
+	defer finishedSig.Signal()
+	a.grpcShutdownStarted.Signal()
+	a.grpcServer.GracefulStop()
```

Comment:
Could this finish and trigger `finishedSig` before `runStopperOnRequest` finishes? 🤔

Reply:
Good idea. Check the latest commit.

Reply:
I like the approach of making `Stop` blocking proposed by Simon.